.github/workflows/release-node.yml

```diff
@@ -54,44 +65,33 @@ jobs:
           max_attempts: 3
           retry_wait_seconds: 60 # wait between retries
           command: cd ${{ matrix.image-name }} && npm i --dry-run
-      - 
-        name: Build image with node version and latest tag
-        uses: docker/build-push-action@v2
-        if: ${{ env.RELEASE_TAG == 'latest' }}
-        with:
-          context: ./${{ matrix.image-name }}
-          file: ./${{ matrix.image-name }}/Dockerfile
-          build-args: NODE_VERSION=${{ env.NODE_VERSION }}
-          load: true
-          tags: apify/actor-${{ matrix.image-name }}:${{ env.RELEASE_TAG }},apify/actor-${{ matrix.image-name }}:${{ env.NODE_VERSION }}
       -
-        name: Build image with release tag
+        name: Build and tag image
         uses: docker/build-push-action@v2
-        if: ${{ env.RELEASE_TAG != 'latest' }}
         with:
           context: ./${{ matrix.image-name }}
           file: ./${{ matrix.image-name }}/Dockerfile
-          build-args: NODE_VERSION=${{ env.NODE_VERSION }}
+          build-args: NODE_VERSION=${{ matrix.node-version }}
           load: true
-          tags: apify/actor-${{ matrix.image-name }}:${{ env.RELEASE_TAG }}
- 
+          tags: ${{ steps.prepare-tags.outputs.result }}
       -
         name: Test image with release tag
+        if: ${{ matrix.node-version == env.NODE_LATEST }}
         run: docker run apify/actor-${{ matrix.image-name }}:${{ env.RELEASE_TAG }}
       -
         name: Test image with node version tag
-        if: ${{ env.RELEASE_TAG == 'latest' }}
-        run: docker run apify/actor-${{ matrix.image-name }}:${{ env.NODE_VERSION }}
+        if: ${{ env.RELEASE_TAG == 'latest'}}
+        run: docker run apify/actor-${{ matrix.image-name }}:${{ matrix.node-version }}
+      -
+        name: Test image with node version tag with release tag
+        if: ${{ env.RELEASE_TAG != 'latest'}}
+        run: docker run apify/actor-${{ matrix.image-name }}:${{ matrix.node-version }}-${{ env.RELEASE_TAG }}
```

Comment:
Do we really need the 3 tests here? We built only a single image in the matrix, does not make much sense to have 3 test steps. For node-version == NODE_LATEST && RELEASE_TAG == 'latest', we would needlessly run two tests.

We can extract one of the tags from the ${{ steps.prepare-tags.outputs.result }}. Either we can export an object from the prepare script and then use the [`fromJSON` function](https://docs.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions#fromjson).
```json
{
    "firstTag": "apify/actor-node:16",
    "joinedTags": "apify/actor-node:16, ..."
}
```

Or maybe we can return an array from the script and use the [`join` function](https://docs.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions#join) in the `tags: ${{ join(steps.prepare-tags.outputs.result) }}` and then just the first item in the array in the tests.

It's confusing to have the three tests here.